### ISSUE TITLE
[FEATURE] Upgrade: Do not fail pre-drain or post-drain jobs

### ISSUE THREAD

**Issue by bk201** [harvester/harvester#3073]
**Is your feature request related to a problem? Please describe.**

Failing pre-drain or post-drain jobs is not a good idea at this moment because:
- Job pods' log might be deleted. It's hard to debug and provide support.
- Cluster is in provisioning (provisioningv2), and it's hard for users to "resume" an upgrade.

**Describe the solution you'd like**

Do not fail the jobs by:
- Either set a very big BackoffLimit (not recommended because logs might be missing).
- Wrap the script with a wrapper.
- Other methods.

**Describe alternatives you've considered**

**Additional context**

**Comment by starbops** [comment]
Question: when we have such a wrapper, i.e. `do_upgrade_node.sh`, is it still necessary/adequate to implement **endless** retry mechanisms inside the real upgrade scripts like bringing up the repo VM or downloading files using curl?

**Comment by TachunLin** [comment]
Verified **fixed** on master-5a18d059-head (11/08). Close this issue.

### Result
When we set the first node to fail in post-drain according to the steps in https://github.com/harvester/harvester/pull/3102#issuecomment-1303769203 

Then we check the post-drain job of the first node, it will continuously prompt the log of retrying in 10 minutes. 
Which make the upgrade dashboard of node1 post-drain don't display `Reach the backoff limit` error message 

### Test Information 
* Test Environment: 4 nodes Harvester bare machines
* Harvester version: master-5a18d059-head (11/08)

### Verify Steps
1. Prepare a 4 nodes v1.0.3 Harvester cluster
2. Enable network on available NIC
3. Create two available VLANs
4. Upgrade from v1.0.3 to master-5a18d059-head
5. After the harvester upgraded in apply-manifest stage, the server-version is changed 
6. Make the first post-drain failed by running 
  ```
  node-0:~ # touch /tmp/elemental
  node-0:~ # mount --bind /tmp/elemental /usr/bin/elemental
  ```
7. Wait for upgrade proceed to first node in pre-drain and post-drain status 
8. Check the post-drain job log 
9. Check the upgrade dashboard status of the first node
